azure-keyvault/azure/keyvault/aio/secrets/_client.py

```diff
@@ -16,8 +16,26 @@
 
 from ...secrets._models import Secret, DeletedSecret, SecretAttributes
 
+# TODO: update all returns and raises
 
 class SecretClient:
+    """The SecretClient class defines a high level interface for managing secrets in the specified vault.
+    
+    :param credentials:  A credential or credential provider which can be used to authenticate to the vault,
+        a ValueError will be raised if the entity is not provided
+    :type credentials: azure.authentication.Credential or azure.authentication.CredentialProvider
+    :param str vault_url: The url of the vault to which the client will connect,
+        a ValueError will be raised if the entity is not provided
+    :param ~azure.core.configuration.Configuration config:  The configuration for the SecretClient
+    
+    Example:
+        .. literalinclude:: ../tests/test_examples_keyvault.py
```

Comment:
Wasn't sure if the test example would have a different approach when doing this so kept this to refer the original documentation.

Reply:
Should be fine, I doubt we'll want an async init.